src/app/clients/edit/[id]/page.tsx

```diff
@@ -1,23 +1,57 @@
-import { Metadata } from "next";
+"use client";
 
-import React from "react";
-import { getSupabaseServerComponentClient } from "@/supabaseServer";
+import React, { useEffect, useState } from "react";
+import supabase from "@/supabaseClient";
 import ClientForm from "@/app/clients/form/ClientForm";
 import { Errors, FormErrors } from "@/components/Form/formFunctions";
 import autofill from "@/app/clients/edit/[id]/autofill";
 import { fetchClient, fetchFamily } from "@/common/fetch";
+import { Schema } from "@/databaseUtils";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { useRouter } from "next/navigation";
+import ErrorPage from "@/app/error";
 
 interface EditClientsParameters {
     params: { id: string };
 }
 
-const EditClients: ({ params }: EditClientsParameters) => Promise<React.ReactElement> = async ({
+const EditClients: ({ params }: EditClientsParameters) => React.ReactElement = ({
     params,
 }: EditClientsParameters) => {
-    const supabase = getSupabaseServerComponentClient();
-    const clientData = await fetchClient(params.id, supabase);
-    const familyData = await fetchFamily(clientData.family_id, supabase);
-    const initialFields = autofill(clientData, familyData);
+    const [clientData, setClientData] = useState<Schema["clients"] | null>(null);
+    const [familyData, setFamilyData] = useState<Schema["families"][] | null>(null);
+    const [error, setError] = useState<Error | null>();
+    const router = useRouter();
+
+    useEffect(() => {
+        if (params.id) {
+            fetchClient(params.id, supabase)
+                .then((client) => {
+                    setError(null);
+                    setClientData(client);
+                })
+                .catch((error) => {
+                    setError({ name: error, message: "Unable to fetch client data" });
+                    logErrorReturnLogId("error fetching client data in client edit modal");
+                });
+        }
+    }, [params.id]);
```

Comment:
nitpick: empty line for separation
```suggestion
    }, [params.id]);

```